tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/im2latex_evaluator.py

```diff
@@ -0,0 +1,306 @@
+        for presenter, metric_result in zip(result_presenters, self._metrics_results):
+            presenter.write_result(metric_result, ignore_results_formatting)
+
+    def release(self):
+        self.model.release()
+        self.launcher.release()
+
+    def reset(self):
+        self.metric_executor.reset()
+        self.model.reset()
+
+    @staticmethod
+    def get_processing_info(config):
+        module_specific_params = config.get('module_config')
+        model_name = config['name']
+        dataset_config = module_specific_params['datasets'][0]
+        launcher_config = module_specific_params['launchers'][0]
+        return (
+            model_name, launcher_config['framework'], launcher_config['device'],
+            launcher_config.get('tags'),
+            dataset_config['name']
+        )
+
+
+class BaseModel:
+    def __init__(self, network_info, launcher, default_model_suffix):
+        self.default_model_suffix = default_model_suffix
+        self.network_info = network_info
+
+    def predict(self, idenitifers, input_data):
+        raise NotImplementedError
+
+    def release(self):
+        pass
+
+    def automatic_model_search(self, network_info):
+        model = Path(network_info['model'])
+        if model.is_dir():
+            is_blob = network_info.get('_model_is_blob')
+            if is_blob:
+                model_list = list(model.glob('*{}.blob'.format(self.default_model_suffix)))
+                if not model_list:
+                    model_list = list(model.glob('*.blob'))
+            else:
+                model_list = list(model.glob('*{}.xml'.format(self.default_model_suffix)))
+                blob_list = list(model.glob('*{}.blob'.format(self.default_model_suffix)))
+                if not model_list and not blob_list:
+                    model_list = list(model.glob('*.xml'.format(self.default_model_suffix)))
+                    blob_list = list(model.glob('*.blob'.format(self.default_model_suffix)))
+                    if not model_list:
+                        model_list = blob_list
+            if not model_list:
+                raise ConfigError('Suitable model for {} not found'.format(self.default_model_suffix))
+            if len(model_list) > 1:
+                raise ConfigError('Several suitable models for {} found'.format(self.default_model_suffix))
+            model = model_list[0]
+        if model.suffix == '.blob':
+            return model, None
+        weights = network_info.get('weights', model.parent / model.name.replace('xml', 'bin'))
+
+        return model, weights
+
+
+def create_recognizer(model_config, launcher, suffix):
+    """Creates encoder and decoder element for Im2LaTeX model
+
+    Args:
+        model_config ([type]): [description]
+        launcher ([type]): [description]
+        suffix ([type]): [description]
+
+    Raises:
+        ValueError: [description]
+
+    Returns:
+        [type]: [description]
+    """
+    launcher_model_mapping = {
```

Comment:
not sure that you need factory-like class initialization if you do not plan to support other frameworks for your model

Reply:
Actually this function is used to crate both encoder and decoder, but you are right, functionality is excess